### PR TITLE
SNOW-907021: Add tests to verify that structured array and maps return the same me…

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
…tadata.

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-907021

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Comment by sfc-gh-stan** [comment]
Why do we need these tests?

File: test/integ/test_structured_types.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            {'a': current_timestamp()::timestamp_tz}::map(string, timestamp_tz),
+            {'a': current_date()}::map(string, date),
+            {'a': current_time()}::map(string, time),
+            {'a': False}::map(string, boolean),
+            {'a': 'b'::variant}::map(string, variant not null),
+            {'a': {'c': 1}}::map(string, object)
+            """
+        )
+        cur.execute(sql)
+        cur.execute(sql)
```
Comment (sfc-gh-aling) [inline test/integ/test_structured_types.py:60]: is the twice execution on purpose?

**Comment by sfc-gh-aling** [comment]
there's a lint error:

trim trailing whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

Fixing test/integ/test_structured_types.py

**Comment by sfc-gh-aling** [comment]
you can skip the old driver test by adding the following lines to the head
```python
import pytest
pytestmark = pytest.mark.skipolddriver  # old test driver tests won't run this module
```

**Comment by sfc-gh-yixie** [comment]
> you can skip the old driver test by adding the following lines to the head
> 
> ```python
> import pytest
> pytestmark = pytest.mark.skipolddriver  # old test driver tests won't run this module
> ```

Added

**Comment by sfc-gh-yixie** [comment]
@sfc-gh-aling All tests pass after I enable the parameters in GCP and Azure.

**Comment by sfc-gh-yixie** [comment]
> Why do we need these tests?

Verify that the new structured types return the same object/array type back to connector so it doesn't break the existing clients (connector and snowpark).

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**